pkg/api/v1/clients/kube/resource_client.go

```diff
@@ -377,6 +392,20 @@ func (rc *ResourceClient) exist(ctx context.Context, namespace, name string) boo
 
 }
 
+func (rc *ResourceClient) convertCrdToInvalidResource(resourceCrd *v1.Resource) (resources.Resource, error) {
+	resourceCrd.Status.State = core.Status_Invalid
+	_, err := rc.crdClientset.ResourcesV1().Resources(resourceCrd.Namespace).Update(resourceCrd)
+	if err != nil {
+		return nil, err
```

Comment:
if any of these errors happen we need to make sure that we don't update the snapshot.

i.e. we must guarantee that if an object exists in storage, it will also exist in the snapshot.